### PR TITLE
move deformable detr safe loading code

### PR DESCRIPTION
**Pull request by HenryL27**
something something lockfile something something deduplication

### CODE REVIEW

**Review by alexaryn — COMMENTED** [review]
I suspect the `to()` needs to be inside the lock.  Without the `to()` I don't think it knows what device to target.  Without that, I don't see how it could do JITC for the correct GPU instruction set.

File: lib/sycamore/sycamore/utils/model_load.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+if TYPE_CHECKING:
+    from transformers import DeformableDetrForObjectDetection
+
+_DETR_LOCK_FILE = f"{Path.home()}/.cache/Aryn-Detr.lock"
+
+
+@requires_modules("transformers", "local_inference")
+def load_deformable_detr(model_name_or_path, device) -> "DeformableDetrForObjectDetection":
+    """Load deformable detr without getting concurrency issues in
+    jit-ing the deformable attention kernel.
```
Comment (alexaryn) [inline lib/sycamore/sycamore/utils/model_load.py:17]: It really should be `jitc`.

**Review by alexaryn — APPROVED** [review]
Seems good to me.  Hope it works.

File: lib/sycamore/sycamore/utils/model_load.py (hunk 1 above)
Comment (HenryL27) [reply lib/sycamore/sycamore/utils/model_load.py]: thank you for the commit message setup

**Pull request merged**